perspective.el

```diff
@@ -139,6 +119,37 @@ After BODY is evaluated, frame parameters are reset to their original values."
   (window-configuration (current-window-configuration))
   (point-marker (point-marker)))
 
+(defmacro persp-current-buffers ()
```

Comment:
Just curious, why make this a macro?

Reply:
Because of `setf` below... If it is a function, `setf` doesn't work (presumably because you get a copy?).

Reply:
Oh, I didn't notice the use of `setf`. It doesn't work with a function because `setf` does magic things and expects a `place` list (https://www.gnu.org/software/emacs/manual/html_node/elisp/Setting-Generalized-Variables.html).